app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -138,6 +140,12 @@ export default function createRPCMethodTrackingMiddleware({
     // keys for the various events in the flow.
     const eventType = EVENT_NAME_MAP[method];
 
+    const isDisabledEthSignAdvancedSetting =
+      method === MESSAGE_TYPE.ETH_SIGN &&
+      res.error?.code === errorCodes.rpc.methodNotFound;
+
+    const isDisabledRequest = isDisabledEthSignAdvancedSetting;
```

Comment:
hey, reason for duplicating the variable ?

Reply:
heyy, I did this for readability down the line

Reply:
hey, I did this for readability down the line. hope it helps, unless it's just me - In that case, happy to change it back